### DISCUSSION TITLE
`exports.browser` in package.json is not automatically picked up on web target

### DISCUSSION THREAD

**Discussion by fregante** [webpack/webpack#14164] [Q&A]
# Bug report

Webpack supports the [`browser` alias](https://webpack.js.org/configuration/resolve/#resolvealiasfields) but it doesn't apply it to exports map.

**What is the current behavior?**

This pattern is supported

```json
{
	"main": "./index.js",
	"browser": "./browser.js"
}
```

but this pattern isn't:

```json
{
	"exports": {
		".": "./index.js",
		"./browser": "./browser.js"
	}
}
```

**If the current behavior is a bug, please provide the steps to reproduce.**

You can check out this folder: https://github.com/fregante/webpack-sandbox/tree/main/exports-browser

It also includes the webpack output (log and dist)

This is part of the output file that demoes which of the 2 files are imported for the "exports map" and "old style" imports respectively:

```js
(() => {
	var e = {
			507: () => {
				console.log('field: exports["."]');
			},
			933: () => {
				console.log('field: browser');
			},
		},
		r = {};
[clipped]
```

**What is the expected behavior?**

`exports.browser` should be picked up just like `browser` is
**Other relevant information:**
webpack 5.31.0
webpack-cli 4.6.0
Node.js version: v16.4.2
Operating System: macOS 11.5

**Comment by alexander-akait** [comment]
Please wait the support, no need to duplicate issues, you already reported and it is investigated

**Reply by alexander-akait** [reply]
Answer https://github.com/webpack/webpack/pull/12693#issuecomment-914257393

**Reply by fregante** [reply]
This was not a question, this was a bug report.

**Reply by alexander-akait** [reply]
There is no bug, please read how `exports` works, it is just path/URL, not declaration field

**Reply by alexander-akait** [reply]
If you have `"./another-name-for-browser": "./browser.js"`, is it code for browser or not? Nobody knows
